app/oauth.go

```diff
@@ -246,7 +246,7 @@ func (a *App) GetOAuthAccessTokenForCodeFlow(clientId, grantType, redirectUri, c
 		var authData *model.AuthData
 		result := <-a.Srv.Store.OAuth().GetAuthData(code)
 		if result.Err != nil {
-			return nil, model.NewAppError("GetOAuthAccessToken", "api.oauth.get_access_token.expired_code.app_error", nil, "", http.StatusInternalServerError)
+			return nil, model.NewAppError("GetOAuthAccessToken", "api.oauth.get_access_token.expired_code.app_error", nil, "", http.StatusBadRequest)
```

Comment:
If I'm not able to get the auth data based on the received code, it can be a regular error, in this case I think is Bad Request. And it shouldn't be an internal server error. We still can generate internal server errors, but normally when we try to save something and we can't. But I think that is the expected behavior.